test/functional/mempool_updatefromblock.py

```diff
@@ -52,73 +48,49 @@ def transaction_graph_test(self, size, n_tx_to_mine=None, start_input_txid='', e
+            vsize = node.getrawmempool(True)[tx['txid']]['vsize']
+            tx_size.append(vsize)
 
+            tx_count = i + 1
             if tx_count in n_tx_to_mine:
                 # The created transactions are mined into blocks by batches.
-                self.log.info('The batch of {} transactions has been accepted into the mempool.'.format(len(self.nodes[0].getrawmempool())))
-                block_hash = self.generate(self.nodes[0], 1)[0]
+                self.log.info('The batch of {} transactions has been accepted into the mempool.'.format(len(node.getrawmempool())))
+                block_hash = self.generate(node, 1)[0]
                 if not first_block_hash:
                     first_block_hash = block_hash
-                assert_equal(len(self.nodes[0].getrawmempool()), 0)
+                assert_equal(len(node.getrawmempool()), 0)
                 self.log.info('All of the transactions from the current batch have been mined into a block.')
             elif tx_count == size:
                 # At the end all of the mined blocks are invalidated, and all of the created
                 # transactions should be re-added from disconnected blocks to the mempool.
-                self.log.info('The last batch of {} transactions has been accepted into the mempool.'.format(len(self.nodes[0].getrawmempool())))
+                self.log.info('The last batch of {} transactions has been accepted into the mempool.'.format(len(node.getrawmempool())))
                 start = time.time()
-                self.nodes[0].invalidateblock(first_block_hash)
+                node.invalidateblock(first_block_hash)
                 end = time.time()
-                assert_equal(len(self.nodes[0].getrawmempool()), size)
+                assert_equal(len(node.getrawmempool()), size)
                 self.log.info('All of the recently mined transactions have been re-added into the mempool in {} seconds.'.format(end - start))
 
         self.log.info('Checking descendants/ancestors properties of all of the in-mempool transactions...')
         for k, tx in enumerate(tx_id):
             self.log.debug('Check transaction #{}.'.format(k))
-            entry = self.nodes[0].getmempoolentry(tx)
+            entry = node.getrawmempool(True)[tx['txid']]
             assert_equal(entry['descendantcount'], size - k)
             assert_equal(entry['descendantsize'], sum(tx_size[k:size]))
             assert_equal(entry['ancestorcount'], k + 1)
             assert_equal(entry['ancestorsize'], sum(tx_size[0:(k + 1)]))
 
     def run_test(self):
         # Use batch size limited by DEFAULT_ANCESTOR_LIMIT = 25 to not fire "too many unconfirmed parents" error.
-        self.transaction_graph_test(size=100, n_tx_to_mine=[25, 50, 75])
+        self.transaction_list_test(size=100, n_tx_to_mine=[25, 50, 75])
 
 
 if __name__ == '__main__':
     MempoolUpdateFromBlockTest().main()
+    
```

Comment:
There is a whitespace here, you have to remove it to pass the linter.

Reply:
Thanks. Fixed. Don't know why the linter now complains about the spelling errors in unchanged files.

^---- failure generated from test/lint/lint-python.sh
src/random.h:92: occuring ==> occurring
src/rpc/blockchain.cpp:794: nWe ==> new
src/util/syscall_sandbox.cpp:128: creat ==> create
test/functional/data/rpc_decodescript.json:81: ba ==> by, be
test/functional/data/rpc_decodescript.json:84: ba ==> by, be
^ Warning: codespell identified likely spelling errors. Any false positives? Add them to the list of ignored words in test/lint/lint-spelling.ignore-words.txt

Reply:
Cool, now you should squash the commits. See: https://github.com/bitcoin/bitcoin/blob/master/CONTRIBUTING.md#squashing-commits

Reply:
Thanks. Done.

Reply:
The problem of the linter now is "test/functional/mempool_updatefromblock.py:12:1: F401 'decimal.Decimal' imported but unused" - this PR removes all uses.